src/d/actor/d_a_obj_mknjd.cpp

```diff
@@ -3,216 +3,1144 @@
+
+            switch (mActionIdx) {
+                case ACT_SETGOAL:
+                    doCutEnd = true;
+                    break;
+                case ACT_SETANGLE:
+                    doCutEnd = true;
+                    break;
+                case ACT_INPUT:
+                    if (talk(1) == 0x12) {
+                        doCutEnd = true;
+                    }
+                    break;
+                case ACT_WAIT:
+                    doCutEnd = true;
+                    break;
+                case ACT_BREAK:
+                    if (daObjMknjD_break() == true) {
+                        if (strcmp(dComIfGp_getStartStageName(), "Ekaze") == 0 || strcmp(dComIfGp_getStartStageName(), "Edaichi") == 0) {
+                            mDoAud_bgmStart(0x80000000 | JA_BGM_JABOO_CAVE);
+                        }
+                        else if (m043E == true) {
+                            mDoAud_bgmStart(0x80000000 | JA_BGM_D_WIND);
+                        }
+                        else {
+                            mDoAud_bgmStart(0x80000000 | JA_BGM_D_EARTH);
+                        }
+
+                        if (mSmokeCBs[2].getEmitter() != NULL) {
+                            mSmokeCBs[2].end();
+                        }
+                        if (mSmokeCBs[3].getEmitter() != NULL) {
+                            mSmokeCBs[3].end();
+                        }
+
+                        doCutEnd = true;
+                    }
+                    break;
+                case ACT_LESSON:
+                    u32 talkResult = talk(1);
+
+                    if (talkResult == 0x12 || talkResult == 0x15) {
+                        doCutEnd = true;
+                    }
+                    break;
+                case ACT_TACT:
+                    if (m0504 == false) {
+                        doCutEnd = true;
+                    }
+                    else {
+                        dComIfGp_event_onEventFlag(8);
+                    }
+                    break;
+                default:
+                    doCutEnd = true;
+                    break;
+            }
+
+            if (doCutEnd) {
+                dComIfGp_evmng_cutEnd(staffIdx);
+            }
+        }
+    }
 }
 
 /* 00001348-00001400       .text manage_friend_draw__10daObjMknjDFi */
-void daObjMknjD::manage_friend_draw(int) {
-    /* Nonmatching */
+void daObjMknjD::manage_friend_draw(int i_param1) {
+    u16 medliProc = PROC_NPC_MD;
+    fopAc_ac_c* judgeResult = static_cast<fopAc_ac_c*>(fopAcIt_Judge(fpcSch_JudgeForPName, &medliProc));
+
+    if (judgeResult != NULL) {
+        if (i_param1 == 1) {
+            u32 prio = fpcLf_GetPriority(judgeResult);
+            fopDwTg_ToDrawQ(&judgeResult->mDwTg, prio);
+        }
+        else {
+            fopDwTg_DrawQTo(&judgeResult->mDwTg);
+        }
+    }
+
+    u16 makarProc = PROC_NPC_CB1;
+    judgeResult = static_cast<fopAc_ac_c*>(fopAcIt_Judge(fpcSch_JudgeForPName, &makarProc));
```

Comment:
im pretty sure there's an inline like `fopAcM_SearchByID` or something like that that makes these kind of calls look a little prettier

Reply:
looks like its `fopAc_ac_c* fopAcM_SearchByName(s16 proc_id)`